datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -112,14 +111,13 @@ object DatasetComparisonJob {
     )
   }
 
-  private def writeMetricsToFile(result: ComparisonResult, fileName: String): Unit = {
-    new File(fileName).mkdirs()
-    val writer = new PrintWriter(s"$fileName/_METRICS")
-    try {
-      writer.write(result.getJsonMetadata)
-    } finally {
-      writer.close()
-    }
+  private def writeMetricsToFile(result: ComparisonResult, fileName: String)(implicit sparkSession: SparkSession): Unit = {
+    val path = new Path(s"$fileName/_METRICS")
+    val fs = FileSystem.get(sparkSession.sparkContext.hadoopConfiguration)
+
+    val v = fs.create(path)
+    try { v.write(result.getJsonMetadata.getBytes) }
```

Comment:
`All this is optional`
Just a personal taste. I try to avoid one-liners like this. Harder to place a breakpoint. Although in IDEA it is still quite easy.
```scala
try {
    v.write(result.getJsonMetadata.getBytes)
}
```

But overall, I'd do things a little differently. Java & Scala have concepts of Input/Output streams and Readers/Writers. The difference is that Input/Output streams are designed for binary data while readers and writers are designed for text data. Readers and Writers have more control over text encoding, for instance.
So I'd write something like this (haven't tested it):
```scala
try {
    val pw = new PrintWriter(v, true)
    pw.print(result.getJsonMetadata)
    pw.close
}
```

Reply:
But `.getBytes()` is fine too, even easier to read. Just add the charset:
```scala
v.write(result.getJsonMetadata.getBytes(StandardCharsets.UTF_8))
```

Reply:
This is not optional. This is a great lecture. Thank you.